arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -785,50 +1095,56 @@ void RocksDBMetaCollection::revisionTreeSummary(VPackBuilder& builder, bool from
 }
 
 void RocksDBMetaCollection::revisionTreePendingUpdates(VPackBuilder& builder) {
-  
   VPackObjectBuilder obj(&builder);
   
   if (!_logicalCollection.useSyncByRevision()) {
     // empty object
     return;
   }
 
-  {
-    std::unique_lock<std::mutex> guard(_revisionTreeLock);
-    if (_revisionTree == nullptr) {
-      // no revision tree yet
-      return;
-    }
+  std::unique_lock<std::mutex> guard(_revisionTreeLock);
+  if (_revisionTree == nullptr) {
+    // no revision tree yet
+    return;
   }
   
-  uint64_t inserts, removes, truncates;
-  {
-    std::unique_lock<std::mutex> guard(_revisionBufferLock);
+  uint64_t inserts = _revisionInsertBuffers.size();
+  uint64_t removes = _revisionRemovalBuffers.size();
+  uint64_t truncates = _revisionTruncateBuffer.size();
 
-    inserts = _revisionInsertBuffers.size();
-    removes = _revisionRemovalBuffers.size();
-    truncates = _revisionTruncateBuffer.size();
-  }
+  guard.unlock();
   
   obj->add("inserts", VPackValue(inserts));
   obj->add("removes", VPackValue(removes));
   obj->add("truncates", VPackValue(truncates));
 }
   
-void RocksDBMetaCollection::placeRevisionTreeBlocker(TransactionId transactionId) {
+uint64_t RocksDBMetaCollection::placeRevisionTreeBlocker(TransactionId transactionId) {
+  auto& engine =
+      _logicalCollection.vocbase().server().getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+  rocksdb::TransactionDB* db = engine.db();
+  
   // make sure that the global revision tree in _revisionTree does not move beyond
   // the sequence number we get here from RocksDB:
   std::unique_lock<std::mutex> guard(_revisionTreeLock);
-  auto& selector =
-      _logicalCollection.vocbase().server().getFeature<EngineSelectorFeature>();
-  auto& engine = selector.engine<RocksDBEngine>();
-  rocksdb::TransactionDB* db = engine.db();
-  rocksdb::SequenceNumber preSeq = db->GetLatestSequenceNumber();
-  // Since we have the lock above, the revision tree cannot move beyond
-  // this sequence number before we have actually placed the blocker.
-  TRI_ASSERT(preSeq >= _revisionTreeApplied);
-  _meta.placeBlocker(transactionId, preSeq);
-  // This remains true, even if `placeBlocker` has advanced the sequence number
+
+  while (true) {
+    rocksdb::SequenceNumber preSeq = db->GetLatestSequenceNumber();
+    // Since we have the lock above, the revision tree cannot move beyond
+    // this sequence number before we have actually placed the blocker.
+    if (preSeq >= _revisionTreeApplied &&
+        preSeq >= _revisionTreeSerializedSeq &&
+        preSeq >= _meta.countCommitted()) {
```

Comment:
Since we have the invariant that `_revisionTreeSerializedSeq <= _revisionTreeApplied` this could be simplified:
```suggestion
    if (preSeq >= _revisionTreeApplied &&
        preSeq >= _meta.countCommitted()) {
```

Reply:
We decided to keep it as is since it doesn't hurt and we want to finalize this.